src/commands/ReminderCommand.ts

```diff
@@ -0,0 +1,174 @@
+import { Command } from "../Command";
+import { CommandInteraction, TextBasedChannel } from "discord.js";
+import {
+	inlineCode,
+	SlashCommandBooleanOption,
+	SlashCommandBuilder,
+	SlashCommandIntegerOption,
+	SlashCommandStringOption,
+	SlashCommandSubcommandsOnlyBuilder,
+} from "@discordjs/builders";
+import { timeDb } from "../db";
+
+class ReminderCommand extends Command {
+	constructor() {
+		super("reminder");
+	}
+
+	private m_id = 0;
+
+	async elapse(interaction: CommandInteraction, caller: string, message = ""): Promise<void> {
+		const ch = (await interaction.client.channels.fetch(interaction.channelId)) as TextBasedChannel;
+		await ch.send({
+			content: `${caller == "-1" ? "@everyone" : `<@${caller}>`}, Time's up! Message: ${message}`,
+		});
+	}
+
+	async execute(interaction: CommandInteraction): Promise<void> {
+		const subcommand = interaction.options.getSubcommand();
+		switch (subcommand) {
+			case "set": {
+				//variables
+				const second = interaction.options.getInteger("second") || 0;
+				const minute = interaction.options.getInteger("minute") || 0;
+				const hour = interaction.options.getInteger("hour") || 0;
+				const message = interaction.options.getString("message") || "";
+				const callAll = interaction.options.getBoolean("callall") || false;
+				const millisecond = (second + 60 * minute + 60 * 60 * hour) * 1000;
+				const d = Date.now() + millisecond;
+				if (!interaction.guild) {
+					await interaction.reply("This command can only be used inside a guild.");
+					return;
+				}
+				const member = await interaction.guild.members.fetch(interaction.user);
+				let i = interaction.user.id;
+				if (callAll) {
+					if (!member.permissions.has("MENTION_EVERYONE")) {
+						await interaction.reply("You don't have the permission, sorry~");
+						return;
+					}
+					i = (-1).toString();
+				}
+				//call setTimeout
+				timeDb.set(this.m_id, {
+					timeout: setTimeout(() => this.elapse(interaction, i, message), millisecond),
+					destination: d,
+					message: message,
+				});
+
+				await interaction.reply(
+					`Okay, I'll remind you soon~~\nIn the event that you wish to no longer be reminded of this timer, use ${inlineCode(
+						`/reminder delete ${this.m_id}`
+					)}`
+				);
```

Comment:
Please test this though :)
```suggestion
				await interaction.reply(
					`Okay, I'll remind you in ${(hour != 0) ? `${hour} hours`} ${(minute != 0) ? `${minute} minutes`} ${(second != 0) ? `${second} seconds`}
					You can delete this timer with: ${inlineCode(
						`/reminder delete ${this.m_id}`
					)}`
				);
```